service/gcs/stdio/stdio.go

```diff
@@ -151,6 +178,13 @@ func (r *TtyRelay) Wait() {
 
 	// Wait for all users of stdioSet and master to finish before closing them.
 	r.wg.Wait()
+
+	// Given the expected use of wait we cannot increment closing before calling r.wg.Wait()
+	// or all calls to ResizeConsole would fail. However, by calling it after there is a very
+	// small window that ResizeConsole could still get an invalid Fd. We call wait again to
+	// enusure that no ResizeConsole call came in before actualling closing the pty.
+	atomic.StoreInt32(&r.closing, 1)
+	r.wg.Wait()
```

Comment:
>wg [](start = 3, length = 2)

I think you need to use a separate mutex unfortunately; according to the docs, you can't legally increment the wait group once Wait has been called:

> go doc sync.WaitGroup.Add
func (wg *WaitGroup) Add(delta int)
    Add adds delta, which may be negative, to the WaitGroup counter. If the
    counter becomes zero, all goroutines blocked on Wait are released. If the
    counter goes negative, Add panics.

    Note that calls with a positive delta that occur when the counter is zero
    must happen before a Wait. Calls with a negative delta, or calls with a
    positive delta that start when the counter is greater than zero, may happen
    at any time. Typically this means the calls to Add should execute before the
    statement creating the goroutine or other event to be waited for. If a
    WaitGroup is reused to wait for several independent sets of events, new Add
    calls must happen after all previous Wait calls have returned. See the
    WaitGroup example.